### PR TITLE
Domainmodel showcase

### PR DESCRIPTION
**Pull request by emilkrebs**
Added an interactive live preview for the domainmodel language using d3.js

image
image

### CODE REVIEW

**Comment by emilkrebs** [comment]
Had to cherry-pick and create a new PR for eclipse ECA, here it is.

File: hugo/assets/scripts/domainmodel/domainmodel-tools.ts [hunk 1]
```diff
@@ -1,5 +1,141 @@
...
+                tags: ['supertype'],
+                children: superType?.features.map(f => getFeatureTreeNode(f) )
+            }] : features;
+
+            return children;
+        }
+        
+        return {
+            ...entity,
+            children: getCildren(entity)
```
Comment (montymxb) [inline hugo/assets/scripts/domainmodel/domainmodel-tools.ts:55]: Typo here, `getChildren`

File: hugo/assets/scripts/domainmodel/domainmodel-tools.ts [hunk 2]
```diff
@@ -1,5 +1,141 @@
...
+            return {
+                name: `${feature.name}${feature.many ? '[]' : ''}`,
+                $type: feature.$type,
+                tags: feature.many ? ['many'] : [],
+                children: [getDataTypeTreeNode(feature.type.ref)]
+            }
+        }
+
+        // get all children of an Entity (including the supertype)
+        const getCildren = (e: Entity): TreeNode[] => {
```
Comment (montymxb) [inline hugo/assets/scripts/domainmodel/domainmodel-tools.ts:39]: Typo

File: hugo/assets/scripts/domainmodel/domainmodel-tools.ts [hunk 3]
```diff
@@ -63,4 +199,4 @@ export const syntaxHighlighting =  {
             { regex: /[\/\*]/, action: {"token":"comment"} },
         ],
     }
-} as monaco.languages.IMonarchLanguage;
\ No newline at end of file
+};
```
Comment (montymxb) [inline hugo/assets/scripts/domainmodel/domainmodel-tools.ts:202]: New line

**Review by montymxb — APPROVED** [review]
Nice! Tested locally and the results look excellent, really happy with the visualization. There are a few little points regarding a typo & new line, but I would consider those non-blocking. With those in place it would be nice, but not required 🙂 .

**Pull request merged**